tests/testthat/test-iv-robust.R

```diff
@@ -106,6 +117,123 @@ test_that("iv_robust matches AER + ivpack", {
     as.matrix(clubsandCR0wo)
   )
 
+  # Rank-deficiency
+  # HC0
+  dat$x1_c <- dat$x
+  ivdefr <- iv_robust(y ~ x + x1_c| z + z2, data = dat, se_type = "HC0")
+  ivdef <- ivreg(y ~ x + x1_c| z + z2, data = dat)
+  ivdefse <- robust.se(ivdef)
+
+  expect_equal(
+    ivdefr$coefficients,
+    ivdef$coefficients
+  )
+
+  expect_equivalent(
+    as.matrix(tidy(ivdefr)[1:2, c("coefficients", "se", "p")]),
+    ivdefse[, c(1, 2, 4)]
+  )
+
+  # Also works as instrument
+  ivdefri <- iv_robust(y ~ z + z2| x + x1_c, data = dat, se_type = "HC0")
+  ivdefi <- ivreg(y ~ z + z2| x + x1_c, data = dat)
+  ivdefsei <- robust.se(ivdefi)
+
+  expect_equal(
+    ivdefri$coefficients,
+    ivdefi$coefficients
+  )
+
+  expect_equivalent(
+    as.matrix(tidy(ivdefri)[1:2, c("coefficients", "se", "p")]),
+    ivdefsei[, c(1, 2, 4)]
+  )
+
+  # Stata
+  ivdefclr <- iv_robust(y ~ x + x1_c | z + z2, data = dat, clusters = clust, se_type = "stata")
+  ivdefcl <- ivreg(y ~ x + x1_c | z + z2, data = dat)
+  ivdefclse <- cluster.robust.se(ivdefcl, clusterid = dat$clust)
+
+  expect_equal(
+    ivdefclr$coefficients,
+    ivdefcl$coefficients
+  )
+
+  expect_equivalent(
+    as.matrix(tidy(ivdefclr)[1:2, c("coefficients", "se")]),
+    ivdefclse[, c(1, 2)]
+  )
```

Comment:
Suggest making a helper function that does the repeated expect_equals for you - eg 

```r
    expect_equal_models <- function(model1, model2) {
      expect_equal(model1$coefficients, model2$coefficients)
      expect_equal(tidy(model1$coefficients), tidy(model2$coefficients))


```


etc.

Reply:
It's a good idea given how much of this there is, but the main problem is depending on my comparison I use different columns I expect to be the same, and they are named different things across packages.